pkg/splunk/enterprise/clustermaster.go

```diff
@@ -137,16 +142,24 @@ func ApplyClusterMaster(client splcommon.ControllerClient, cr *enterpriseApi.Clu
 	}
 
 	// create or update a regular service for the cluster manager
-	err = splctrl.ApplyService(client, getSplunkService(cr, &cr.Spec.CommonSplunkSpec, SplunkClusterMaster, false))
+	err = splctrl.ApplyService(client, getSplunkService(cr, &cr.Spec.CommonSplunkSpec, SplunkClusterManager, false))
 	if err != nil {
 		return result, err
 	}
 
 	// create or update statefulset for the cluster manager
-	statefulSet, err := getClusterMasterStatefulSet(client, cr)
+	statefulSet, err := getClusterManagerStatefulSet(client, cr)
 	if err != nil {
 		return result, err
 	}
+
+	//make changes to respective mc configmap when changing/removing mcRef from spec
+	extraEnv, err := VerifyCMisMultisite(cr, namespaceScopedSecret)
+	err = validateMonitoringConsoleRef(client, statefulSet, extraEnv)
+	if err != nil {
+		return result, err
+	}
+
 	clusterMasterManager := splctrl.DefaultStatefulSetPodManager{}
```

Comment:
If `clusterMasterManager` is a local variable non-CRD dependent, we could change it to avoid using "Master". Maybe clusterManagerPod or something of that sort?